app/assets/javascripts/dashboard/dashboard.js

```diff
@@ -439,43 +444,58 @@ MinionPoller = {
     var isWorker = MinionPoller.selectedNodes.indexOf(minion.id) !== -1;
     var isUnused = !isMaster && !isWorker;
 
-    var masterHtml;
     var masterChecked = '';
-    var minionHtml;
     var minionChecked = '';
+    var disabledHtml = '';
+    if (State.pendingRemovalMinionId != null){
+      disabledHtml = 'disabled="disabled"';
+    }
 
     if (isMaster) {
       masterChecked = 'checked';
     }
 
-    masterHtml = '\
-      <input name="roles[master][]" id="roles_master_' + minion.id +
+    var masterHtml = '\
+      <input hidden name="roles[master][]" id="roles_master_' + minion.id +
       '" value="' + minion.id + '" type="checkbox" title="Select node as master" ' + masterChecked + '>';
 
     if (isWorker) {
       minionChecked = 'checked';
     }
 
-    minionHtml = '<input name="roles[worker][]" id="roles_worker_' + minion.id +
+    var minionHtml = '\
+      <input hidden name="roles[worker][]" id="roles_worker_' + minion.id +
       '" value="' + minion.id + '" type="checkbox" title="Select node for bootstrapping" ' + minionChecked + '>';
-
     var roleHtml = '\
-      <td class="role-column">' +
+      <td class="role-column" '+ disabledHtml +' id="role-column-'+ minion.minion_id +'">' +
```

Comment:
No need of `disabled` attribute in `td` element.

Reply:
ok